### DISCUSSION TITLE
Preserve the original file name when using name format different than Name

### DISCUSSION THREAD

**Discussion by Adaliciously** [diced/zipline#247] [Ideas]
I would like for zipline to save the original filename, so even when using a Format different than NAME, saving the file will result in file being saved with original filename. It could be possible with HTTP Content-Disposition header.

**Comment by TacticalTechJay** [comment]
Feature requests were moved to discussions. Could you start a thread there and close this issue?

**Reply by TacticalTechJay** [reply]
~~or this could be converted to a discussion but alright~~

**Comment by TacticalTechJay** [comment]
What would be the benefits from this?

**Reply by Adaliciously** [reply]
preserving the original file name while keeping the file/url randomized

**Comment by TacticalTechJay** [comment]
If the files were saved with the original file name, two files that have identical names AND extensions can't exist

**Reply by Adaliciously** [reply]
they don't need to be! the files (and urls) would still be randomly generated, however a proper filename would be used while downloading the file (using content-disposition)

**Reply by Adaliciously** [reply]
![image](https://user-images.githubusercontent.com/7670940/209097990-743e954d-9dbc-4589-b921-a3e12b97509c.png)
This is how lolisafe which I'm coming from handles it

**Reply by Adaliciously** [reply]
![image](https://user-images.githubusercontent.com/7670940/209100072-c639d57b-e72d-47ea-ac1d-043df42bf99a.png)
This website illustrates kinda well how content disposition works:
https://demo.borland.com/testsite/download_testpage.php

**Reply by diced** [reply]
I do find it weird why you'd want to set it to a different file name (through Content-Disposition), instead of just using the original name format. Since I'm not opposing this, you may see this feature added in the near future.

**Reply by Adaliciously** [reply]
> I do find it weird why you'd want to set it to a different file name (through Content-Disposition), instead of just using the original name format. Since I'm not opposing this, you may see this feature added in the near future.

since I would like to have the benefit of randomized short urls, while in the meantime having the ability to store the original file name.

For example: I have a config file with a very long name
I want to upload it, and download it on other machine - but if I want to use the randomised url i will have to rename the file manually

**Reply by TacticalTechJay** [reply]
Hmm.. With seeing this, how about I raise you this #248. It's what I tried adding, would this match your intended idea?

**Comment by Adaliciously** [comment]
Also, sorry for not making it into discussion, I totally forgot that this feature exists on github as I'm used to issues.

**Comment by diced** [comment]
Added in https://github.com/diced/zipline/commit/4ea1775f2ca8ea4a89c45637164c9e413d677954!

**Reply by Adaliciously** [reply]
thank you so much!

**Reply by Adaliciously** [reply]
one more thing, is there a way to set this to be default? via .env or any other way?

**Reply by Adaliciously** [reply]
Found a bug with this, if you upload a file without the option enabled, the file will be saved as null
![image](https://user-images.githubusercontent.com/7670940/212639731-04ee0d78-e6bb-4371-a612-ef807d220804.png)

**Reply by Adaliciously** [reply]
![image](https://user-images.githubusercontent.com/7670940/212639819-1dbfdecf-3c35-4472-807b-76441e5e36e3.png)
